src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -86,8 +86,14 @@ public async Task<AnalyzerResult> AnalyzeAsync()
                     Contract.ThrowIfFalse(unused.Count == 0);
                 }
 
-                // check whether instance member is used inside of the selection
-                var instanceMemberIsUsed = IsInstanceMemberUsedInSelectedCode(dataFlowAnalysisData);
+                var thisParameterBeingRead = dataFlowAnalysisData.ReadInside.FirstOrDefault(s => IsThisParameter(s)) as IParameterSymbol;
+                var isThisParameterWritten = dataFlowAnalysisData.WrittenInside.Any(s => IsThisParameter(s));
+
+                var instanceMemberIsUsed = thisParameterBeingRead is object || isThisParameterWritten;
+                var shouldBeReadOnly = !isThisParameterWritten
+                    && thisParameterBeingRead?.Type is ITypeSymbol thisType
```

Comment:
i don't understand the purpose of the check.  this feels more convoluted than just

```c#
var shouldBeReadOnly = !isThisParameterWritten
  && thisParameterBeingRead != null
  && thisParameterBeingRead.Type.TypeKind == TypeKind.Struct 
  && thisParameterBeingRead.Type.IsReadOnly;
```

Reply:
Sure, I can change it to that.

Reply:
Done. See what you think of the final factoring of this.

Reply:
i like it!  I have to be start looking for these pattern opportunities myself!